pycheribuild/projects/cross/apache.py

```diff
@@ -0,0 +1,150 @@
+                                              # XXX: This might be yes on Linux
+                                              ac_cv_func_setpgrp_void="no",
+                                              ac_cv_struct_rlimit="yes",
+                                              ac_cv_func_sem_open="yes",
+                                              apr_cv_process_shared_works="yes",
+                                              apr_cv_mutex_robust_shared="yes",
+                                              # XXX: This might be yes on Linux
+                                              apr_cv_tcp_nodelay_with_cork="no",
+                                              )
+
+        self.run_cmd("./buildconf", cwd=self.source_dir)
+        super().configure(**kwargs)
+
+
+class BuildApache(CrossCompileAutotoolsProject):
+    target = "apache"
+    project_name = "apache"
+
+    repository = GitRepository("https://github.com/CTSRD-CHERI/apache-httpd.git",
+                               default_branch="2.4.x-cheri")
+
+    dependencies = ["apr", "pcre"]
+
+    native_install_dir = DefaultInstallDir.IN_BUILD_DIRECTORY
+    cross_install_dir = DefaultInstallDir.ROOTFS_LOCALBASE
+
+    def configure(self, **kwargs):
+        self.configure_args.extend([
+            "--enable-layout=FreeBSD",
+            "--enable-http",
+            "--enable-mod-ssl",
+            "--with-expat=" + str(self.install_dir),
+            "--with-pcre=" + str(BuildPcre.get_install_dir(self)),
+            "--with-apr=" + str(BuildApr.get_install_dir(self)),
+            ])
+        if self.build_type.is_debug:
+            self.configure_args.append("--enable-debugger-mode")
+
+        # The cross-compile always assumes this is true (so one
+        # can't normally cross-compile Apache), and it's a fatal
+        # error, so just lie always.
+        self.configure_environment.update(ap_cv_void_ptr_lt_long="no")
+
+        # mod_ssl doesn't use -L from --with-expat
+        self.LDFLAGS.append("-L" + str(self.install_dir / "lib"))
+
+        if not self.compiling_for_host() and False:
+            # Can't determine these when cross-compiling
+            self.configure_environment.update(ac_cv_file__dev_zero="yes",
+                                              ac_cv_mmap__dev_zero="yes",
+                                              # XXX: This might be yes on Linux
+                                              ac_cv_func_setpgrp_void="no",
+                                              ac_cv_struct_rlimit="yes",
+                                              ac_cv_func_sem_open="yes",
+                                              apr_cv_process_shared_works="yes",
+                                              apr_cv_mutex_robust_shared="yes",
+                                              # XXX: This might be yes on Linux
+                                              apr_cv_tcp_nodelay_with_cork="no",
+                                              )
+
```

Comment:
Maybe add the setup()/configure() split here as well?

Reply:
Will do.